### ISSUE TITLE
Problem/Suggestion regarding Penatration.

### ISSUE THREAD

**Issue by BlazingInFury** [pWn3d1337/Techguns#21]
Hey, so recently I've been playing your mod and i've really enjoyed it, it's honestly one of my favorite mods so far and it's very well built, however I have a request to suggest to improve on Multiplayer Servers. Is there any way you can set it so that there is a config option or something similar to disable gun penetration?  It'd really help as Armor Penetration is quite powerful and unbalanced, especially if players don't use the armor you provide.  Unless there already is a solution to that, it'd help if you can explain it to me. Other than that, thank you for your assistance!

**Comment by pWn3d1337** [comment]
No, there is no real solution yet besideds using techguns armors. The config only allows general damage multipliers for NPCs or PvP. It's a bit problematic, because without penetration some of the better weapons would be very weak. Then other mods add super overpowered armours. 

But Armor Penetration is nothing new. The problem is how vanilla minecraft works. Armor only reduces damage from "normal" damage. If you set something to magic damage it completely bybasses armor, and yes some mods do that. With modded armors (ISpecialArmor Interface from forge you can interfere with that in any way you want and change calculations at will.). I tried to improve the system by adding damage types and making penetration a float between 0 and 1 instead of binary no penetration/full penetration. It does work well within the mod, but gives problems with other mods trying to do their own damage tricks. 

What exactly do you think is the problem? PvP penetration or Techguns NPCs penetrating non-Techguns armors?

I can try too add config options that disables penetration for NPCs or in PvP.

**Comment by BlazingInFury** [comment]
Thank you for helping me understand how your system works, makes everything a whole lot clearer. I would like a config option that disables penetration for PvP if you can. Main reasoning is that the modpack (As well as the server) is quite balanced in our classes/armor options.  (I'm a Moderator/Admin an a Star Wars modpack on the Technic Launcher. And we're currently re-making/Updating our pack to be really well made with planets, weapons etc.) Also, is the issue with the TechGuns tab not syncing up when teleportingto other dimensions fixed? It'd be great to know. Many thanks for your reply.

**Comment by pWn3d1337** [comment]
I think I will add both options. 
As for the tab syncing, I'm looking into it, I had that problem on a dedicated server, but when I hosted a dedicated server on my own PC and connected to it, it does work. I will test again with a second PC, but I'm not sure what caused this. The code to sync the player is there.

**Comment by BlazingInFury** [comment]
I would assume the issue could be caused by what the server is attempting to do. We had a similar problem with that using the Plugin known as MultiWorld, which could perhaps be the issue. Anyways, thank you for your help, I greatly appreciate it. And I can't wait for the update to come!

**Comment by BlazingInFury** [comment]
Also, if you can, can you leave a message here so I can be notified of the update? I would greatly appreciate it.

**Comment by pWn3d1337** [comment]
I just noticed that penetration does not even work against players... I think I need to check the damage calculations again, there also seems to be a bug also where armor does not work correctly.

**Comment by BlazingInFury** [comment]
May I ask which armor's?

**Comment by pWn3d1337** [comment]
All techguns armors seem to have a problem with penetration where the calculation is wrong, but I fixed that already, will be in next version.

**Comment by BlazingInFury** [comment]
Nice, can't wait till the next update. Will it add more content or will it simply be bug fixes?
